### PR TITLE
irv changes

### PR DESCRIPTION
**Pull request by Harindhar10**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/IRV.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
+from deepchem.models.torch_models.torch_model import TorchModel
+from deepchem.models.losses import SigmoidCrossEntropy
+import torch
+import torch.nn as nn
+from typing import Callable, Optional
+
+if torch.cuda.is_available():
```
Comment (rbharath) [inline deepchem/models/torch_models/IRV.py:7]: Please remove this code; this is dangerous and has a global effect!

File: deepchem/models/torch_models/IRV.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        self.K = K
+        self.penalty = penalty
+
+        # Initialize weights and biases
+        self.V = nn.Parameter(torch.tensor([0.01, 1.], dtype=torch.float32))
+        self.W = nn.Parameter(torch.tensor([1., 1.], dtype=torch.float32))
+        self.b = nn.Parameter(torch.tensor([0.01], dtype=torch.float32))
+        self.b2 = nn.Parameter(torch.tensor([0.01], dtype=torch.float32))
+
+    def forward(self, inputs):
```
Comment (rbharath) [inline deepchem/models/torch_models/IRV.py:93]: Docstring and type annotation!

File: deepchem/models/torch_models/IRV.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        outputs = [
+            torch.cat([outputs2, outputs], dim=2),
+            logits[0] if len(logits) == 1 else torch.cat(logits, dim=1)
+        ]
+        return outputs
+
+
+class Slice(nn.Module):
+    """ Choose a slice of input on the last axis given order,
+    Suppose input x has two dimensions,
```
Comment (rbharath) [inline deepchem/models/torch_models/IRV.py:133]: Add a usage example

File: deepchem/models/torch_models/IRV.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        self.axis = axis
+
+    def forward(self, inputs: torch.Tensor) -> torch.Tensor:
+        slice_num = self.slice_num
+        axis = self.axis
+        return inputs.index_select(axis, torch.tensor([slice_num],
+                                                      device=device))
+
+
+class MultitaskIRVClassifier(TorchModel):
```
Comment (rbharath) [inline deepchem/models/torch_models/IRV.py:157]: Docstring and usage example!

Comment (rbharath) [inline deepchem/models/torch_models/IRV.py:157]: Add this to the .rst docs

**Review by rbharath — COMMENTED** [review]
A number of changes needed (see below). For docs:

- Add layers to layers.rst
- Add model to models.rst
- Update model cheatsheet

Also make sure to remove dangerous global update of device

File: deepchem/models/torch_models/IRV.py (hunk 4 above)
Comment (Harindhar10) [reply deepchem/models/torch_models/IRV.py:151]: added

Comment (Harindhar10) [reply deepchem/models/torch_models/IRV.py:151]: added

File: deepchem/models/torch_models/IRV.py (hunk 3 above)
Comment (Harindhar10) [reply deepchem/models/torch_models/IRV.py:127]: added

File: deepchem/models/torch_models/IRV.py (hunk 2 above)
Comment (Harindhar10) [reply deepchem/models/torch_models/IRV.py:87]: added

File: deepchem/models/torch_models/IRV.py (hunk 1 above)
Comment (Harindhar10) [reply deepchem/models/torch_models/IRV.py]: removed

**Pull request closed without merging**